lib/videojs-playlist-ui.js

```diff
@@ -1,25 +1,188 @@
-
-    // TODO: write some amazing plugin code
-  };
-
-  // register the plugin
-  videojs.plugin('playlistUi', playlistUi);
-})(window, window.videojs);
+import * as playlist from 'videojs-playlist';
+
+const defaults = {
+  className: 'vjs-playlist'
+};
+
+const createThumbnail = function(thumbnail) {
+  if (!thumbnail) {
+    let placeholder = document.createElement('div');
+    placeholder.className = 'vjs-playlist-thumbnail';
+    return placeholder;
+  }
+
+  let picture = document.createElement('picture');
+  picture.className = 'vjs-playlist-thumbnail';
+
+  if (typeof thumbnail === 'string') {
+    // simple thumbnails
+    let img = document.createElement('img');
+    img.src = thumbnail;
+    picture.appendChild(img);
+  } else {
+    // responsive thumbnails
+
+    // additional variations of a <picture> are specified as
+    // <source> elements
+    for (let i = 0; i < thumbnail.length - 1; i++) {
+      let variant = thumbnail[i];
+      let source = document.createElement('source');
+      // transfer the properties of each variant onto a <source>
+      for (let prop in variant) {
+        source[prop] = variant[prop];
+      }
+      picture.appendChild(source);
+    }
+
+    // the default version of a <picture> is specified by an <img>
+    let variant = thumbnail[thumbnail.length - 1];
+    let img = document.createElement('img');
+    for (let prop in variant) {
+      img[prop] = variant[prop];
+    }
+    picture.appendChild(img);
+  }
+  return picture;
+};
+
+videojs.PlaylistMenuItem = videojs.Component.extend({
+  init: function(player, options) {
+    if (!options.item) {
+      throw new Error('Cannot construct a PlaylistMenuItem without an item option');
+    }
+    // stub out the element so Component doesn't construct one
+    options.el = true;
+    videojs.Component.call(this, player, options);
+    this.el_ = this.createEl(options.item);
+
+    this.item_ = options.item;
+    this.emitTapEvents();
+
+    this.on(['click', 'tap'], (event) => {
+      const currentIndex = player.playlist().indexOf(this.item_);
+      if (currentIndex >= 0) {
+        player.playlist.currentItem(currentIndex);
+      }
+    });
+  },
+  createEl: function(item) {
+    let li = document.createElement('li');
+    li.className = 'vjs-playlist-item';
+
+    // Thumbnail image
+    li.appendChild(createThumbnail(item.thumbnail));
+
+    // Duration
+    if (item.duration) {
+      let duration = document.createElement('time');
+      let time = videojs.formatTime(item.duration);
+      duration.className = 'vjs-playlist-duration';
+      duration.setAttribute('datetime', 'PT0H0M' + item.duration + 'S');
+      duration.appendChild(document.createTextNode(time));
+      li.appendChild(duration);
+    }
+
+    // Name and description
+    let name = document.createElement('cite');
+    name.className = 'vjs-playlist-name';
+    name.appendChild(document.createTextNode(item.name || this.localize('Untitled Video')));
```

Comment:
The app developer would have to register localizations with video.js for any `item.name` they planned on using. Seems more straight forward to just tell them to make sure their video object is correctly localized before loading it to me. Make sense?